kuma/wiki/jinja2/wiki/react_document.html

```diff
@@ -83,264 +82,69 @@
-                  {{ _("This is an experimental page.") }}
-                </div>
-              {% endif %}
-
-              {% if kumascript_errors %}
-                {% include 'wiki/includes/kumascript_errors.html' %}
-              {% endif %}
-
-              {% if document.parent_id and document.current_revision %}
-                {% if document.current_revision.localization_in_progress %}
-                  {% if not settings.MAINTENANCE_MODE and document.current_revision.translation_age >= 10 %}
-                    <div class="overheadIndicator translationInProgress"><p>{% trans translate_url=doc_edit_url %}
-                      This translation is incomplete. Please help <a href="{{ translate_url }}">translate this article</a> from English.
-                    {% endtrans %}</p></div>
-                  {% else %}
-                    <div class="overheadIndicator translationInProgress"><p>
-                      {{ _('This translation is in progress.') }}
-                    </p></div>
-                  {% endif %}
-                {% endif %}
-              {% endif %}
-              {% if not settings.MAINTENANCE_MODE and fallback_reason %}
-                <div id="doc-pending-fallback" class="warning">
-                  <p><bdi>{% trans help_link=help_link, locale=settings.LOCALES[request.LANGUAGE_CODE].native, parent_language=document.language %}
-                    Our volunteers haven't translated this article into <bdi>{{ locale }}</bdi> yet.
-                    <a href="{{ help_link }}" rel="nofollow">Join us and help get the job done!</a><br>
-                    You can also read the article in <a href="{{ doc_abs_url }}">{{ parent_language }}</a>.
-                  {% endtrans %}</bdi></p>
-                </div>
-              {% endif %}
-
-              <!-- just the article content -->
-              <article id="wikiArticle">
-                {% if not fallback_reason %}
-                  {% if not document_html %}
-                    {{ _("This article doesn't have any content yet.") }}
-                  {% else %}
-                    {{ body_html|safe }}
-                  {% endif %}
-                {% elif fallback_reason == 'no_translation' %}
-                  {{ body_html|safe }}
-                {% elif fallback_reason == 'translation_not_approved' %}
-                  {{ document.parent.html|safe }}
-                {% else %}
-                  {{ _("This article doesn't have approved content yet.") }}
-                {% endif %}
-              </article>
-
-              <!-- contributors -->
-              <div class="wiki-block contributors">
-                <h2 class="offscreen">{{ _('Document Tags and Contributors') }}</h2>
-                {% set tags = document.all_tags_name %}
-                {% include "wiki/includes/document_tag.html" %}
+  <!-- Content will go here -->
+  <div id="react-container">
+    <script type="base64-encoded-json">{{ document_api_data }}</script>
```

Comment:
Here's where I hardcode the document API json payload so that react has all the data it needs to render without making a fetch() call.